earthpy/spatial.py

```diff
@@ -346,22 +347,34 @@ def plot_bands(arr, title=None, cmap="Greys_r",
     --------
     >>>import earthpy.spatial as es
     ...
-    ...titles = ["Red Band", "Green Band", "Blue Band",
-    ...          "Near Infrared (NIR) Band"]
+    ...im = np.random.randint(10, size=(2, 4, 5))
+    ...titles = ["Red Band", "Green Band"]
     ...
     ...# Plot all bands of a raster tif
-    ...es.plot_bands(naip_image,
+    ...es.plot_bands(im,
     ...              title=titles,
     ...              figsize=(12,5),
     ...              cols=2)
     """
+
+    try:
+        arr.ndim
+    except AttributeError:
+        "Input arr should be a numpy array"
+
+
+    if title:
+        if (arr.ndim == 2) and (len(title) > 1):
+            raise ValueError("""You have provided more than one title for a
```

Comment:
May be worth being super explicit about what is expected here, e.g., "N titles were provided for a single band array, but plot_bands() expects exactly one title for a single band array", pasting in the number of titles that were provided for "N".

Reply:
fixed!!